src/screens/AddressesScreen.tsx

```diff
@@ -61,20 +62,58 @@ const AddressesScreen = ({ navigation }: ScreenProps) => {
   const confirmedTransactions = useAppSelector((state) => selectConfirmedTransactions(state, [currentAddressHash]))
   const pendingTransactions = useAppSelector((state) => selectPendingTransactions(state, [currentAddressHash]))
 
-  const onScrollEnd = (index: number) => {
-    setCurrentAddressHash(addressHashes[index])
-    setAreButtonsDisabled(false)
-  }
-
   useFocusEffect(
     useCallback(() => {
       if (currentAddressHash) setAreButtonsDisabled(false)
     }, [currentAddressHash])
   )
 
-  if (!currentAddress) return null
+  useEffect(() => {
+    if (mainAddress) setCurrentAddressHash(mainAddress)
+  }, [mainAddress])
+
+  const onAddressCardsScrollEnd = useCallback(
+    (index: number) => {
+      if (index < addressHashes.length) setCurrentAddressHash(addressHashes[index])
+      setAreButtonsDisabled(false)
+    },
+    [addressHashes]
+  )
+
+  const onAddressCardsScrollStart = useCallback(() => setAreButtonsDisabled(true), [])
+
+  const onLayoutCarouselItem = useCallback(
+    (event: LayoutChangeEvent) => setHeightCarouselItem(event.nativeEvent.layout.height),
+    []
+  )
+
+  const renderAddressCard = useCallback(
+    ({ item }: { item: string }) => (
+      <View onLayout={onLayoutCarouselItem} key={item}>
+        <AddressCard addressHash={item} />
+      </View>
+    ),
+    [onLayoutCarouselItem]
+  )
 
-  const onLayoutCarouselItem = (event: LayoutChangeEvent) => setHeightCarouselItem(event.nativeEvent.layout.height)
+  const navigateToSendScreen = useCallback(
+    () => navigation.navigate('SendScreen', { addressHash: currentAddressHash }),
+    [currentAddressHash, navigation]
+  )
+
+  const navigateToReceiveScreen = useCallback(
+    () => navigation.navigate('ReceiveScreen', { addressHash: currentAddressHash }),
+    [currentAddressHash, navigation]
+  )
+
+  const navigateToEditAddressScreen = useCallback(
+    () => navigation.navigate('EditAddressScreen', { addressHash: currentAddressHash }),
+    [currentAddressHash, navigation]
+  )
+
+  const closeQRCodeModal = useCallback(() => setIsQrCodeModalOpen(false), [])
```

Comment:
I fear that we may be going a bit too far with the use of `useCallback`, as this can be counter-productive in term of performances (and code complexity). We should make sure that this is needed.

Some interesting articles:
- https://kentcdodds.com/blog/usememo-and-usecallback
- https://dmitripavlutin.com/react-usecallback/

Reply:
Thanks, I realized I need to revise this PR after what I learnt with the links you shared in the explorer PR!